### DISCUSSION TITLE
Function components cannot be given refs

### DISCUSSION THREAD

**Discussion by mateusborba** [react-hook-form/react-hook-form#10539] [Q&A] [answered]
Hy guys! I'm having this warning in the console and I dont know why.
I have a comboBox from headless ui and I put it on a Controller component.
thats the code: 

comboBox:
![comboBoxcode](https://github.com/react-hook-form/react-hook-form/assets/83471193/7fe7e9d9-7d21-43eb-b451-c8e3b6e72a36)

controller: 

![controller](https://github.com/react-hook-form/react-hook-form/assets/83471193/4eeb995c-d379-410c-91d5-c94ba87bb8a4)

**Comment by Moshyfawn** [comment, marked as answer]
You can't simply pass `ref` as a prop in React. You need to wrap your `SearchComboBox` component in [`forwardRef`](https://react.dev/reference/react/forwardRef)
